### PR TITLE
Adds Fetch Data and Read Write Examples

### PR DESCRIPTION
**Pull request by RobDWaller**
This pull requests adds two further basic examples to the examples list, the aim is to introduce concepts such as security, flags and the differences between the runtime API and the standard library.

Examples include:
- Fetch data over the web.
- Read and write files to the file system.

Work contributes further to #6844.

Future PRs may add one final basic example and will also begin tidying up existing advanced examples.

### CODE REVIEW

File: docs/examples/fetch_data.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+**Command:** `deno run --allow-net fetch.ts`
+
+```js
+async function callApi(url: string): Promise<Response> {
+  try {
+    return await fetch(url);
+  } catch (e) {
+    throw new Error(e.message);
+  }
+}
```
Comment (lucacasonato) [inline docs/examples/fetch_data.md:22]: Is this not an unnecessary abstraction? Could this not be removed and replaced with direct calls to `fetch()`?

File: docs/examples/read_write_files.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+ *   {"id": 5, "name": "Devika", "age": 11}
+ * ]
+ */
+```
+
+## Write
+
+The Deno runtime API allows developers to write text to files via the
+`writeTextFile()` method. It just requires a file path and text string. On
+success the method will return an empty promise.
```
Comment (lucacasonato) [inline docs/examples/read_write_files.md:81]: > On success the method will return an empty promise.

The method returns a promise which resolves when the file was successfully written.

File: docs/examples/read_write_files.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+To execute the code the `deno run` command needs the unstable flag and both the
+write and read flags.
+
+**Command:** `deno run --allow-write --allow-read --unstable write.ts`
+
+```js
+import {
+  ensureFileSync,
+  writeJsonSync,
+} from "https://deno.land/std@0.65.0/fs/mod.ts";
```
Comment (lucacasonato) [inline docs/examples/read_write_files.md:122]: Can you change the imports to use `std@$STD_VERSION`? This special string is replaced in the frontend with the correct std version.

File: docs/examples/fetch_data.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Fetch Data
+
+When building any sort of web application developers will usually need to
+retrieve data from somewhere else on the web. This works no differently in Deno
+than in any other JavaScript application, just call the the `fetch()` method.
```
Comment (lucacasonato) [inline docs/examples/fetch_data.md:5]: Maybe we should also link to MDN for further reference?

**Review by lucacasonato — COMMENTED** [review]
Looking good :+1:

**Comment by RobDWaller** [comment]
@lucacasonato thanks for the suggestions will make the improvements you suggest. 👍

File: docs/examples/fetch_data.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+/**
+ * Output: JSON Data
+**/
+callApi("https://api.github.com/users/denoland")
+  .then((response) => {
+    return response.json();
+  })
+  .then((json) => console.log(json));
```
Comment (zlianon) [inline docs/examples/fetch_data.md:31]:
```typescript
.then(console.log);

File: docs/examples/fetch_data.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Output: HTML Data
+**/
+callApi("https://deno.land/")
+  .then((response) => {
+    return response.text();
+  })
+  .then((text) => console.log(text));
```
Comment (zlianon) [inline docs/examples/fetch_data.md:40]:
``` typescript
.then(console.log);

**Comment by RobDWaller** [comment]
@lucacasonato I've made the changes suggested, thank you.

Can I just confirm though `std@$STD_VERSION` is just a place holder for documentation, right?

**Comment by lucacasonato** [comment]
> Can I just confirm though `std@$STD_VERSION` is just a place holder for documentation, right?

Yup. It is just a placeholder for docs.

**Review by lucacasonato — APPROVED** [review]
LGTM - thanks Rob

**Pull request merged**